File: src/features/ai-platform/infra/commands/sessions.ts
import { invoke } from '@tauri-apps/api/core';

export interface SessionAssetRecord {
  id: string;
  name: string;
  sessionType: string;
  group?: string;
  target: string;
  projectDir?: string;
  summary: string;
  resumeCommand?: string;
  createdAt?: string;
  expiredAt?: string;
  proxyEnabled: boolean;
  tunnelEnabled: boolean;
}

export interface SessionGroupRecord {
  name: string;
  count: number;
}

export interface SessionsSnapshot {
  sessions: SessionAssetRecord[];
  groups: SessionGroupRecord[];
  deeplinkTemplate: string;
  source: string;
}

export function getAiPlatformSessionsSnapshot() {
  return invoke<SessionsSnapshot>('ai_platform_get_sessions_snapshot');
}

export function deleteAiPlatformSessionRecord(sessionId: string) {
  return invoke<SessionsSnapshot>('ai_platform_delete_session_record', { sessionId });
}

File: src-tauri/src/ai_platform/domain/sessions.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SessionAssetRecord {
    pub id: String,
    pub name: String,
    pub session_type: String,
    pub group: Option<String>,
    pub target: String,
    pub project_dir: Option<String>,
    pub summary: String,
    pub resume_command: Option<String>,
    pub created_at: Option<String>,
    pub expired_at: Option<String>,
    pub proxy_enabled: bool,
    pub tunnel_enabled: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SessionGroupRecord {
    pub name: String,
    pub count: usize,
}